src/main/java/org/spongepowered/api/text/format/TextFormat.java

```diff
@@ -117,9 +117,27 @@ public final TextFormat color(TextColor color) {
         return new TextFormat(this.style, color);
     }
 
+    /**
+     * Returns a new {@link TextFormat} that combines this and the given format.
+     * The given format takes higher priority than this one. Due to this the
+     * color will only fallback to this one if the given format's color is
+     * {@link TextColors#NONE} or {@link TextColors#RESET}. Styles are combined
+     * using {@link TextStyle#and(TextStyle...)}.
+     *
+     * @param format The format to merge
+     * @return A new {@link TextFormat}
+     */
+    public final TextFormat merge(TextFormat format) {
+        TextColor color = format.getColor();
+        if (color == TextColors.NONE || color == TextColors.RESET) {
```

Comment:
How should `TextColors.RESET` be treated like?
Is this the correct merge behaviour?